src/components/base/outline-modal/outline-modal.ts

```diff
@@ -0,0 +1,251 @@
+    }
+
+    return id;
+  }
+
+  async open(): Promise<void> {
+    if (!this.isOpen) {
+      this.isOpen = true;
+
+      await this.updateComplete;
+
+      this._focusOnElement();
+
+      this._trapFocus();
+
+      this.dispatchEvent(new CustomEvent('opened'));
+    }
+  }
+
+  async close(): Promise<void> {
+    if (this.isOpen) {
+      this.isOpen = false;
+
+      await this.updateComplete;
+
+      this.dispatchEvent(new CustomEvent('closed'));
+
+      this.triggerElement.focus();
+    }
+  }
+
+  @query('#trigger')
+  private triggerElement!: HTMLDivElement;
+
+  private _handleTriggerKeydown(event: KeyboardEvent): void {
+    if (event.key === 'Enter') {
+      // This prevents a focused element from also triggering.
+      // For example, the modal opens and the "accept" button is focused and then triggered and the modal closes.
+      event.preventDefault();
+
+      this.open();
+    }
+  }
+
+  private _handleOverlayClick(event: MouseEvent): void {
+    // Only trigger if we click directly on the event that wants to receive the click.
+    if (event.target === event.currentTarget) {
+      this.close();
+    }
+  }
+
+  private _handleOverlayKeydown(event: KeyboardEvent): void {
+    if (event.key === 'Escape') {
+      this.close();
+    }
+  }
+
+  // For some reason on the `Docs` tab of Storybook, the `click` event for the close button doesn't work with the `Enter` key without also watching the `keyup` event. This isn't the case on the `Canvas` tab.
+  private _handleCloseKeydown(event: KeyboardEvent): void {
+    if (event.key === 'Enter') {
+      this.close();
+    }
+  }
+
+  @query('#close')
+  private closeElement!: HTMLDivElement;
+
+  @property({ type: String })
+  elementToFocusSelector?: string | undefined;
+
+  private _focusOnElement(): void {
+    let elementToFocus: HTMLElement = this.closeElement;
+
+    if (this.elementToFocusSelector !== undefined) {
+      const attributeDefinedElementToFocus = this.querySelector(
+        this.elementToFocusSelector
+      ) as HTMLElement | null;
+
+      if (attributeDefinedElementToFocus !== null) {
+        elementToFocus = attributeDefinedElementToFocus;
+      }
+    }
+
+    if (this.elementToFocusSelector === undefined) {
+      const automaticallySelectedElementToFocus = this.querySelector(
+        focusableElementSelector
+      ) as HTMLElement | null;
+
+      if (automaticallySelectedElementToFocus !== null) {
+        elementToFocus = automaticallySelectedElementToFocus;
+      }
+    }
+
+    elementToFocus.focus();
+  }
+
+  private _trapFocus(): void {
+    // We will use the close button as the first focusable element.
+
+    let lastFocusableElement: HTMLElement = this.closeElement;
+
+    const focusableElements: NodeListOf<HTMLElement> = this.querySelectorAll(
+      focusableElementSelector
+    );
+
+    if (focusableElements.length > 0) {
+      lastFocusableElement = focusableElements[focusableElements.length - 1];
+    }
```

Comment:
Would this make more sense?

```typescript
const focusableElements: NodeListOf<HTMLElement> = this.querySelectorAll(focusableElementSelector);

const lastFocusableElement: HTMLElement = focusableElements[focusableElements.length - 1] ?? this.closeElement;
```

This avoids re-assigning to `lastFocusableElement`.